### PR TITLE
Correction in ANN layers

### PR DESCRIPTION
**Pull request by shubham1206agra**
- Added `Layer<MatType>()` in the constructor wherever it is missing.
- Added copy and move constructor in PaddingType layer.
- Fixed SoftmaxType move constructor.

### CODE REVIEW

File: src/mlpack/methods/ann/layer/padding_impl.hpp [hunk 1]
```diff
@@ -34,6 +35,60 @@ PaddingType<MatType>::PaddingType(
...
+PaddingType<MatType>&
+PaddingType<MatType>::operator=(const PaddingType& other)
+{
+  if (this != &other)
+    Layer<MatType>::operator=(other);
+    padWLeft = other.padWLeft;
+    padWRight = other.padWRight;
+    padHTop = other.padHTop;
+    padHBottom = other.padHBottom;
+    totalInMaps = other.totalInMaps;
```
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/padding_impl.hpp:72]:
```suggestion
  if (this != &other)
  {
    Layer<MatType>::operator=(other);
    padWLeft = other.padWLeft;
    padWRight = other.padWRight;
    padHTop = other.padHTop;
    padHBottom = other.padHBottom;
    totalInMaps = other.totalInMaps;
  }
```
Maybe too much Python? :smile:

File: src/mlpack/methods/ann/layer/padding_impl.hpp [hunk 2]
```diff
@@ -34,6 +35,60 @@ PaddingType<MatType>::PaddingType(
...
+PaddingType<MatType>&
+PaddingType<MatType>::operator=(PaddingType&& other)
+{
+  if (this != &other)
+    Layer<MatType>::operator=(std::move(other));
+    padWLeft = std::move(other.padWLeft);
+    padWRight = std::move(other.padWRight);
+    padHTop = std::move(other.padHTop);
+    padHBottom = std::move(other.padHBottom);
+    totalInMaps = std::move(other.totalInMaps);
```
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/padding_impl.hpp:87]:
```suggestion
  if (this != &other)
  {
    Layer<MatType>::operator=(std::move(other));
    padWLeft = std::move(other.padWLeft);
    padWRight = std::move(other.padWRight);
    padHTop = std::move(other.padHTop);
    padHBottom = std::move(other.padHBottom);
    totalInMaps = std::move(other.totalInMaps);
  }
```

File: src/mlpack/methods/ann/layer/softmax_impl.hpp [hunk 3]
```diff
@@ -35,7 +35,7 @@ SoftmaxType<MatType>::SoftmaxType(const SoftmaxType& other) :
 
 template<typename MatType>
 SoftmaxType<MatType>::SoftmaxType(SoftmaxType&& other) :
-    Layer<MatType>(other)
+    Layer<MatType>(std::move(other))
```
Comment (rcurtin) [inline src/mlpack/methods/ann/layer/softmax_impl.hpp:38]: Ahh, nice catch. :+1:

**Review by rcurtin — COMMENTED** [review]
Thank you for catching this!  Just two small things to fix. :)  Also, I know it is a little tedious, but if you don't mind adding a description to the PR, it can make it a little easier to review.

**Comment by shubham1206agra** [comment]
> Thank you for catching this! Just two small things to fix. :) Also, I know it is a little tedious, but if you don't mind adding a description to the PR, it can make it a little easier to review.

Done :)

**Review by rcurtin — APPROVED** [review]
Awesome, thanks again! :rocket:

**Pull request merged**